dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/imports/HarvestedJsonParser.java

```diff
@@ -0,0 +1,577 @@
+
+        Dataset dataset = new Dataset();
+
+        dataset.setAuthority(obj.getString("authority", null) == null ? settingsService.getValueForKey(SettingsServiceBean.Key.Authority) : obj.getString("authority"));
+        dataset.setProtocol(obj.getString("protocol", null) == null ? settingsService.getValueForKey(SettingsServiceBean.Key.Protocol) : obj.getString("protocol"));
+        dataset.setIdentifier(obj.getString("identifier", null));
+
+        DatasetVersion dsv = new DatasetVersion();
+        dsv.setDataset(dataset);
+        dsv = parseDatasetVersion(obj.getJsonObject("datasetVersion"), dsv);
+        List<DatasetVersion> versions = new ArrayList<>(1);
+        versions.add(dsv);
+
+        dataset.setVersions(versions);
+        return dataset;
+    }
+
+    private DatasetVersion parseDatasetVersion(JsonObject obj, DatasetVersion dsv) throws JsonParseException {
+        try {
+
+            String archiveNote = obj.getString("archiveNote", null);
+            if (archiveNote != null) {
+                dsv.setArchiveNote(archiveNote);
+            }
+
+            dsv.setDeaccessionLink(obj.getString("deaccessionLink", null));
+            int versionNumberInt = obj.getInt("versionNumber", -1);
+            Long versionNumber = null;
+            if (versionNumberInt != -1) {
+                versionNumber = (long) versionNumberInt;
+            }
+            dsv.setVersionNumber(versionNumber);
+            dsv.setMinorVersionNumber(Option.of(obj.getString("minorVersionNumber", null)).map(Long::parseLong).getOrNull());
+            // if the existing datasetversion doesn not have an id
+            // use the id from the json object.
+            if (dsv.getId() == null) {
+                dsv.setId(Option.of(obj.getString("id", null)).map(Long::parseLong).getOrNull());
+            }
+
+            String versionStateStr = obj.getString("versionState", null);
+            if (versionStateStr != null) {
+                dsv.setVersionState(DatasetVersion.VersionState.valueOf(versionStateStr));
+            }
+            dsv.setReleaseTime(parseDate(obj.getString("releaseDate", null)));
+            dsv.setLastUpdateTime(parseTime(obj.getString("lastUpdateTime", null)));
+            dsv.setCreateTime(parseTime(obj.getString("createTime", null)));
+            dsv.setArchiveTime(parseTime(obj.getString("archiveTime", null)));
+            dsv.setUNF(obj.getString("UNF", null));
+            // Terms of Use related fields
+            dsv.setTermsOfUseAndAccess(new TermsOfUseAndAccess());
+
+            dsv.setDatasetFields(parseMetadataBlocks(obj.getJsonObject("metadataBlocks")));
+
+            JsonArray filesJson = obj.getJsonArray("files");
+
+            Option<TermsOfUseAndAccess.License> license = parseLicense(obj.getString("license", null));
+            String termsOfUse = obj.getString("termsOfUse", null);
+            Tuple2<Option<TermsOfUseAndAccess.License>, String> oldLicenses = Tuple.of(license, termsOfUse);
```

Comment:
I don't quite understand, I mean I understand that I shouldn't use legacy ENUM's since they are going to be removed. But 'going directly into FileTermsOfUse' is the part I don't get. Do you want me to add legacy Enums in FileTermsOfUse?

Reply:
You're `parseLicense` could return `Option<FileTermsOfUse>` and when assigning it to specific `FileMetadata` we could make `FileMetadata.setTermsOfUse(whatweparsed.createCopy());`